doc/compiler/jitserver/Caching.md

```diff
@@ -0,0 +1,49 @@
+Caching helps alleviate this problem by storing the result of frequent remote calls on the server, so that if the same data is needed again, it can accesss it from the cache, instead of making a remote call.
+
+# Types of caching
+There are 2 types of caching that `JITServer` does: global (in persistent memory) and per-compilation (on the heap). Both types of caching are done on per-client basis, that is, if multiple clients are connected to the same server, they will not share caches, as that would make things very complicated.
+Global caching is done for things that will not change (or very unlikely to change) over the lifetime of a client JVM, e.g. GC mode, IProfiler data for compiled methods.
+Local caching is done for things that are not going to change during the current compilation, but might change in-between compilations or are just unique for each compilation, e.g. resolved methods, IProfiler data for interpreted methods.
+Whenever possible, caching should be done globally, because hit rates will be higher, but one should be careful and make sure that the client data will not actually change.
+
+# Important caches
+## `ClientSessionData`
+Stores all of the globally cached data, for a detailed description read [this](https://github.com/eclipse/openj9/blob/jitaas/doc/compiler/jitaas/ClientSession.md).
+## Caches in `CompilationInfoPerThreadRemote`
+All of the per-compilation caches are stored in `CompilationInfoPerThreadRemote`. Since most caches have very similar structure, i.e. hash map, we added templated methods for working with these caches. The methods are `initializePerCompilationCache`, `cacheToPerCompilationMap`, `getCachedValueFromPerCompilationMap`, `clearPerCompilationCache`. If a new cache is added, it should use these methods. At the end of compilation all local caches need to be reset by adding a call to `clearPerCompilationCache` to `TR::CompilationInfoPerThreadRemote::clearPerCompilationCaches`.
+Some important per-compilation caches:
+### `TR_ResolvedMethodInfoCache`
+This cache stores pointers to resolved methods created for the current compilaton. It is important, because messages requesting the creation of resolved methods are some of the most frequent messages. Unfortunately, persistent caching does not seem possible, since every compilation creates its own resolved methods.
+### `IPTableHeap_t`
+This cache stores IProfiler data for methods whose profiling data might get updated. It is one of the two caches used by IProfiler. The other one is persistent and stores data for already compiled methods, because their interpreter profiling data will definitely not change. This cache stores data for interpreted methods and a method currently being compiled. 
```

Comment:
When referring to "`It is one of the two caches used by IProfiler. The other one is persistent and stores data for already compiled methods`”, could you add a bit more information on the other persistent cache data such as method/class name used by IProfiler?